src/manifest.js

```diff
@@ -0,0 +1,187 @@
+ *        The uri to the media playlist (if media playlist is not from within a master
+ *        playlist)
+ * @param {string} id
+ *        ID to use for the playlist
+ */
+export const setupMediaPlaylist = ({ playlist, uri, id }) => {
+  playlist.id = id;
+
+  if (uri) {
+    // For media playlists, m3u8-parser does not have access to a URI, as HLS media
+    // playlists do not contain their own source URI, but one is needed for consistency in
+    // VHS.
+    playlist.uri = uri;
+  }
+
+  // For HLS master playlists, even though certain attributes MUST be defined, the
+  // stream may still be played without them.
+  // For HLS media playlists, m3u8-parser does not attach an attributes object to the
+  // manifest.
+  //
+  // To avoid undefined reference errors through the project, and make the code easier
+  // to write/read, add an empty attributes object for these cases.
+  playlist.attributes = playlist.attributes || {};
+};
+
+/**
+ * Adds ID, resolvedUri, and attributes properties to each playlist of the master, where
+ * necessary. In addition, creates playlist IDs for each playlist and adds playlist ID to
+ * playlist references to the playlists array.
+ *
+ * @param {Object} master
+ *        The master playlist
+ */
+export const setupMediaPlaylists = (master) => {
+  let i = master.playlists.length;
+
+  while (i--) {
+    const playlist = master.playlists[i];
+
+    setupMediaPlaylist({
+      playlist,
+      id: createPlaylistID(i, playlist.uri)
+    });
+    playlist.resolvedUri = resolveUrl(master.uri, playlist.uri);
+    master.playlists[playlist.id] = playlist;
+
+    // Although the spec states an #EXT-X-STREAM-INF tag MUST have a BANDWIDTH attribute,
+    // the stream can be played without it. Although an attributes property may have been
+    // added to the playlist to prevent undefined references, issue a warning to fix the
+    // manifest.
+    if (!playlist.attributes.BANDWIDTH) {
+      log.warn('Invalid playlist STREAM-INF detected. Missing BANDWIDTH attribute.');
+    }
+  }
+};
+
+/**
+ * Adds resolvedUri properties to each media group.
+ *
+ * @param {Object} master
+ *        The master playlist
+ */
+export const resolveMediaGroupUris = (master) => {
+  forEachMediaGroup(master, (properties) => {
+    if (properties.uri) {
+      properties.resolvedUri = resolveUrl(master.uri, properties.uri);
+    }
+  });
+};
+
+/**
+ * Creates a master playlist wrapper to insert a sole media playlist into.
+ *
+ * @param {Object} media
+ *        Media playlist
+ * @param {string} uri
+ *        The media URI
+ *
+ * @return {Object}
+ *         Master playlist
+ */
+export const masterForMedia = (media, uri) => {
+  const id = createPlaylistID(0, uri);
+  const master = {
+    mediaGroups: {
+      'AUDIO': {},
+      'VIDEO': {},
+      'CLOSED-CAPTIONS': {},
+      'SUBTITLES': {}
+    },
+    uri: window.location.href,
+    playlists: [{
+      uri,
+      id,
+      resolvedUri: uri,
+      attributes: {}
```

Comment:
Should we add back some of the old inline comments from the code this function is based on?